src/Features/VisualBasic/Completion/KeywordRecommenders/Statements/FinallyKeywordRecommender.vb

```diff
@@ -41,5 +41,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.KeywordRecommenders.Stat
 
             Return SpecializedCollections.EmptyEnumerable(Of RecommendedKeyword)()
         End Function
+
+        Private Function IsInCatchOfTry(targetToken As SyntaxToken, tryBlock As TryBlockSyntax) As Boolean
+            Dim parent = targetToken.Parent
+            While parent IsNot tryBlock
```

Comment:
You would hit the tryBlock first--we already know it's an ancestor of targetToken.